src/snapshot_middleware/mod.rs

```diff
@@ -18,30 +18,39 @@ mod toml;
 mod txt;
 mod util;
 
-use std::path::Path;
+use std::{
+    path::{Path, PathBuf},
+    sync::OnceLock,
+};
 
+use anyhow::Context;
 use memofs::{IoResultExt, Vfs};
+use serde::{Deserialize, Serialize};
 
-use crate::snapshot::{InstanceContext, InstanceSnapshot};
+use crate::glob::Glob;
+use crate::snapshot::{InstanceContext, InstanceSnapshot, SyncRule};
 
 use self::{
     csv::{snapshot_csv, snapshot_csv_init},
     dir::snapshot_dir,
     json::snapshot_json,
     json_model::snapshot_json_model,
-    lua::{snapshot_lua, snapshot_lua_init},
+    lua::{snapshot_lua, snapshot_lua_init, ScriptType},
     project::snapshot_project,
     rbxm::snapshot_rbxm,
     rbxmx::snapshot_rbxmx,
     toml::snapshot_toml,
     txt::snapshot_txt,
-    util::PathExt,
 };
 
 pub use self::{project::snapshot_project_node, util::emit_legacy_scripts_default};
 
-/// The main entrypoint to the snapshot function. This function can be pointed
-/// at any path and will return something if Rojo knows how to deal with it.
+static DEFAULT_SYNC_RULES: OnceLock<Vec<SyncRule>> = OnceLock::new();
```

Comment:
If you think it's any better, you can stick this in the `default_sync_rules` function so that the two aren't split by the rest of the file. This might also be good because `DEFAULT_SYNC_RULES` is not meant to be used directly, it's only meant to be used indirectly by calling `default_sync_rules`.